lib/composer.js

```diff
@@ -194,6 +194,10 @@ export default class Composer extends Disposable {
 
     await Promise.all(jobs)
 
+    if (this.shouldCleanAfterBuild()) {
```

Comment:
@oserbetci We are not currently tracking the success/failure of the build. We will need some extra code to make the clean conditional.

Since error messages could come from the log, not just from a returned status code (which will generate a log message), one option would be to add the following to `Logger`.

```javascript
hasPendingErrors () {
  return this._group && this._group.some(message => message.type === 'error')
}
```

Then you could do:

```javascript
if (!latex.log.hasPendingErrors() && this.shouldCleanAfterBuild()) {
```